### DISCUSSION TITLE
Session Proposal Namespace Acceptance

### DISCUSSION THREAD

**Discussion by chrislang1** [WalletConnect/web-examples#30] [Q&A] [answered]
Hi,

I was wondering if it should be possible for a wallet to decline specific networks or methods within a namespace? ie. If the dApp requests Polygon, Optimism and Ethereum, does the user need to accept _all_ of these networks to connect, or can they accept some and decline the others?

Thanks!

**Comment by xzilja** [comment, marked as answer]
Hey @chrislang1 

So, as it stands now you need to satisfy / approve `requiredNamespaces` proposed by dapp, assumption here is that they will only request what's needed for core functionality and if it's not met there might be unexpected side-effects. You can allow "more" than what's required, but never less.

Same applies to `update` method. You can always add more to namespace, but never less than what was in original `requiredNamespaces`.

We'll also work on more docs around namespaces and will be adding more granular errors that should make it super easy to understand these behaviours 👍

**Reply by chrislang1** [reply]
Awesome, thanks for the info @IljaDaderko 😊
